stock_reserve_sale/model/sale.py

```diff
@@ -137,98 +136,116 @@ def _is_stock_reservable(self):
         string='Stock Reservation',
         copy=False)
     is_stock_reservable = fields.Boolean(
-        compute='_is_stock_reservable',
+        compute='_compute_is_stock_reservation',
         readonly=True,
         string='Can be reserved')
 
+    @api.multi
+    def _prepare_stock_reservation(self, date_validity=False, note=False):
+        self.ensure_one()
+
+        try:
+            owner_id = self.stock_owner_id and self.stock_owner_id.id or False
+        except AttributeError:
+            owner_id = False
+            # module sale_owner_stock_sourcing not installed, fine
+
+        return {'product_id': self.product_id.id,
+                'product_uom': self.product_uom.id,
+                'product_uom_qty': self.product_uom_qty,
+                'date_validity': date_validity,
+                'name': "%s (%s)" % (self.order_id.name, self.name),
+                'note': note,
+                'price_unit': self.price_unit,
+                'sale_line_id': self.id,
+                'restrict_partner_id': owner_id,
+                }
+
+    @api.multi
+    def acquire_stock_reservation(self, date_validity=False, note=False):
+        reserv_obj = self.env['stock.reservation'].sudo()
+
+        reservations = reserv_obj.browse()
+        for line in self:
+            if not line.is_stock_reservable:
+                continue
+
+            vals = line._prepare_stock_reservation(
+                date_validity=date_validity, note=note)
+
+            # Place picking_type_id in context. This is required
+            # to make reserve automaticaly find location_id and
+            # location_dest_id
+            pick_type_id = line.order_id.warehouse_id.int_type_id.id
```

Comment:
Why doing this as in https://github.com/OCA/stock-logistics-warehouse/blob/99c8a3add3ab98736d52424633f7197fac1574b0/stock_reserve/model/stock_reserve.py#L96 the default is set ?

Reply:
Sorry, this is the good one : https://github.com/OCA/stock-logistics-warehouse/blob/6c8d4d0e2c5a74a2da6abc68bb2a0834b209fc96/stock_reserve/model/stock_reserve.py#L118

Reply:
It seems that this (`_default_picking_type_id`) method may return incorrect picking type in case of multiple warehouses.

Reply:
So, shouldn't that method be fixed instead of doing another manner as this module depends on it ?

Reply:
At this time we do not use this addon, we have no Odoo9 installations running. Thus i do not plan to add any changes to this PR that require deep dive in reservation logic.

As for `_default_picking_type_id` method - it seems that there is no way to get `warehouse` in this method.